### PR TITLE
Add rest parameter BBEs for function pointers and anonymous functions

### PR DESCRIPTION
**Pull request by irshadnilam**
## Purpose
$title

### CODE REVIEW

File: examples/function-pointers/function_pointers.bal [hunk 1]
```diff
@@ -26,11 +26,28 @@ function getFunctionPointer() returns (function (int, string) returns float) {
     return test;
 }
 
+function functionWithRestValues(int... r) returns int {
+    int total = 0;
+
+    foreach var item in r {
+        total = total + item;
```
Comment (MaryamZi) [inline examples/function-pointers/function_pointers.bal]:
```suggestion
        total += item;
```

File: examples/function-pointers/function_pointers.bal [hunk 2]
```diff
@@ -26,11 +26,28 @@ function getFunctionPointer() returns (function (int, string) returns float) {
...
 public function main() {
     // Value `test` will serve as a function pointer for the `foo` function.
     io:println("Answer: ", foo(10, test));
     io:println("Answer: ", foo(10, getFunctionPointer()));
+
     // A function pointer as a variable.
     function (int, string) returns float f = getFunctionPointer();
+
+    // A function pointer with rest params as a variable
+    function (int...) returns int total= functionWithRestValues;
```
Comment (MaryamZi) [inline examples/function-pointers/function_pointers.bal]:
```suggestion
    function (int...) returns int total = functionWithRestValues;
```

File: examples/function-pointers/function_pointers.bal [hunk 3]
```diff
@@ -26,11 +26,28 @@ function getFunctionPointer() returns (function (int, string) returns float) {
...
+
 public function main() {
     // Value `test` will serve as a function pointer for the `foo` function.
     io:println("Answer: ", foo(10, test));
     io:println("Answer: ", foo(10, getFunctionPointer()));
+
     // A function pointer as a variable.
     function (int, string) returns float f = getFunctionPointer();
+
+    // A function pointer with rest params as a variable
```
Comment (MaryamZi) [inline examples/function-pointers/function_pointers.bal]:
```suggestion
    // A function pointer, with a rest parameter, as a variable.
```

File: examples/anonymous-functions/anonymous_functions.bal [hunk 4]
```diff
@@ -14,6 +14,12 @@ public function main() {
                         };
     io:println("Output: ", anonFunction2("Hello ", "World.!!!"));
 
+    // Defines an anonymous function with rest parameter.
```
[the remark this answers is not in the document]
Comment (irshadnilam) [reply examples/anonymous-functions/anonymous_functions.bal]: Added to existing method.

File: examples/function-pointers/function_pointers.bal [hunk 5]
```diff
@@ -15,7 +15,7 @@ function test(int x, string s) returns float {
     return f;
 }
 
-// A function pointer as a parameter. Uses the `.call()` method to invoke the function using the function pointer.
+// A function pointer as a parameter. A function pointer can be invoked just as invoking a normal function.
```
[the remark this answers is not in the document]
Comment (irshadnilam) [reply examples/function-pointers/function_pointers.bal]: Changed the sentence to suggested.

File: examples/anonymous-functions/anonymous_functions.bal [hunk 6]
```diff
@@ -14,6 +14,12 @@ public function main() {
...
+    var anonFunction3 = function (string... a) returns int {
+                            return  a.length();
```
[the remark this answers is not in the document]
Comment (irshadnilam) [reply examples/anonymous-functions/anonymous_functions.bal]: No longer apply.

File: examples/function-pointers/function_pointers.bal [hunk 7]
```diff
@@ -26,11 +26,28 @@ function getFunctionPointer() returns (function (int, string) returns float) {
     return test;
 }
 
+function functionWithRestValues(int... r) returns int {
```
[the remark this answers is not in the document]
Comment (irshadnilam) [reply examples/function-pointers/function_pointers.bal]: Moved to existing function.

File: examples/function-pointers/function_pointers.bal (hunk 1 above)
Comment (irshadnilam) [reply examples/function-pointers/function_pointers.bal]: No longer apply, followed this in other places.

File: examples/function-pointers/function_pointers.bal (hunk 2 above)
Comment (irshadnilam) [reply examples/function-pointers/function_pointers.bal]: No longer apply,

File: examples/function-pointers/function_pointers.bal (hunk 3 above)
Comment (irshadnilam) [reply examples/function-pointers/function_pointers.bal]: Changed.

File: examples/function-pointers/function_pointers.bal [hunk 8]
```diff
@@ -2,35 +2,40 @@ import ballerina/io;
...
     // Value `test` will serve as a function pointer for the `foo` function.
     io:println("Answer: ", foo(10, test));
     io:println("Answer: ", foo(10, getFunctionPointer()));
+
     // A function pointer as a variable.
-    function (int, string) returns float f = getFunctionPointer();
+    function (string, int...) returns float f = getFunctionPointer();
+
     io:println("Answer: ", foo(10, f));
+
```
Comment (MaryamZi) [inline examples/function-pointers/function_pointers.bal]: Shall we remove this extra new line?

Comment (irshadnilam) [reply examples/function-pointers/function_pointers.bal]: Removed.

File: examples/function-pointers/function_pointers.bal [hunk 9]
```diff
@@ -2,35 +2,39 @@ import ballerina/io;
...
+// A function pointer as a parameter. A function pointer can be invoked similar to how a normal function is invoked.
+function foo(int x, function (string, int...) returns float bar)
              returns float {
-    return x * bar(10, "2");
+    return x * bar("2", 2, 3, 4, 5);
 }
 
 // A function pointer as a return type.
-function getFunctionPointer() returns (function (int, string) returns float) {
+function getFunctionPointer() returns (function (string, int...) returns float) {
```
Comment (anupama-pathirage) [inline examples/function-pointers/function_pointers.bal]: These lines are greater than 80 character width, which will make it not clear with random line breaks at b.io site. So limit the line width to max of 80 chars. Refer for more details : https://github.com/ballerina-platform/ballerina-lang/blob/master/docs/bbe-guidelines/bbe-guidelines.md

Comment (irshadnilam) [reply examples/function-pointers/function_pointers.bal]: Resolved.

**Pull request merged**